### PR TITLE
fix: HTTP|HTTPS|NO_PROXY env variable reading #3055

### PR DESCRIPTION
**Pull request by zeph**
@jannfis tnx for the snippet ;)

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Optional. My organization is added to the README.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Comment by zeph** [comment]
...I got no idea how to fix this Semantic thing; suggestions are more than welcome

File: README.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ Organizations below are **officially** using Argo CD. Please send a PR with your
 1. [Viaduct](https://www.viaduct.ai/)
 1. [Volvo Cars](https://www.volvocars.com/)
 1. [Walkbase](https://www.walkbase.com/)
-
+1. [Whitehat Berlin](https://whitehat.berlin) at [ATU](https://www.atu.de/)/[Mobivia](https://www.mobivia.com/) & [UMI](https://www.linkedin.com/company/urban-mobility-international/)/[WeShare](http://we-share.io/)/[Volkswagen](http://volkswagen.de/)
```
Comment (jannfis) [inline README.md]: Just a question, because I know the corporate dos-and-donts too well: Do you have permission to use these references?

**Review by jannfis — CHANGES_REQUESTED** [review]
Thanks for your PR! 

Regarding the "Semantic PR" check, this affects mostly the title of your PR - it needs to be prefixed with a given "class" for this PR, in this case it would be `fix: ` - please refer to https://github.com/probot/semantic-pull-requests for more details. Also, I'd suggest re-phrasing the title of the PR to something more confident, along the lines of "Enable HTTP proxy support when connecting Helm repositories" or similar - we use the PR titles in the Changelog.

Please also see the comment regarding the references. @alexec @alexmt are mentioning the references OK for you?

File: README.md (hunk 1 above)
Comment (zeph) [reply README.md]: @jannfis I just sent a CYA mail to both... 
clarifying that this is a "Thank you UMI & ATU for sponsoring my work on ArgoCD" and nothing else

**Review by jannfis — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (alexmt) [reply README.md]: just sync up with @edlee2121 about it. We think that this list should not have references. 
They might add themself if they want :) 

@zeph , could you please remove the references and keep just the company name?

**Review by alexmt — CHANGES_REQUESTED** [review]
Thank you for contribution @zeph ! 

Added one comment about references.

**Comment by zeph** [comment]
@alexmt removed as requested, best

**Comment by jannfis** [comment]
@zeph Thanks! The CI was failing, I guess it was a hiccup. Can you please check and re-trigger CI processes, i.e. by pushing an empty commit (`git commit --allow-empty`) to your branch? We will only merge when all checks are passed.

**Review by alexmt — APPROVED** [review]

**Pull request merged**